shared/js/textChatView.jsx

```diff
@@ -225,28 +240,22 @@ loop.shared.views.chat = (function(mozL10n) {
             }
             {
               this.props.messageList.map(function(entry, i) {
-                if (entry.type === CHAT_MESSAGE_TYPES.SPECIAL) {
-                  if (!this.props.showInitialContext) {
-                    return null;
-                  }
-
-                  switch (entry.contentType) {
-                    case CHAT_CONTENT_TYPES.CONTEXT:
-                      return (
-                        <div className="context-url-view-wrapper" key={i}>
-                          <sharedViews.ContextUrlView
-                            allowClick={true}
-                            description={entry.message}
-                            dispatcher={this.props.dispatcher}
-                            thumbnail={entry.extraData.thumbnail}
-                            url={entry.extraData.location} />
-                        </div>
-                      );
-                    default:
-                      console.error("Unsupported contentType",
-                                    entry.contentType);
-                      return null;
-                  }
+                // if we have to show context tile,
+                // return special view
+                if (entry.type === CHAT_MESSAGE_TYPES.SPECIAL &&
+                    entry.contentType === CHAT_CONTENT_TYPES.CONTEXT &&
+                    !!this.props.showInitialContext) {
```

Comment:
If `this.props.showInitialContext` is `false` we're going to render a `TextChatEntry` view but we don't want to render that